pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,585 @@
+  security:
+    citadel:
+      namespace: istio-citadel
+    cert_manager:
+      namespace: istio-cert-manager
+    node_agent:
+      namespace: istio-node-agent
+
+5. Default install with specialized k8s settings for pilot
+
+spec:
+  traffic_management:
+    pilot:
+        k8s:
+          resources:
+            limits:
+              cpu: 444m
+              memory: 333Mi
+            requests:
+              cpu: 222m
+              memory: 111Mi
+          readinessProbe:
+            failureThreshold: 44
+            initialDelaySeconds: 11
+            periodSeconds: 22
+            successThreshold: 33
+
+6. Default install with values.yaml customizations for proxy
+
+spec:
+  traffic_management:
+    proxy:
+      valuesOverrides:
+      - global.proxy.enableCoreDump: true
+      - global.proxy.dnsRefreshRate: 10s
+
+7. Default install with modification to container flag in galley
+
+spec:
+  configuration_management:
+    galley:
+      k8s:
+        overlays:
+        - apiVersion: extensions/v1beta1
+          kind: Deployment
+          name: istio-galley
+          patches:
+          - path: spec.template.spec.containers.[name:galley].command.[--livenessProbeInterval]
+            value: --livenessProbeInterval=123s
+*/
+
+
+message IstioInstall {
+    InstallSpec spec = 3;
+    InstallStatus status = 4;
+}
+
+// InstallSpec defines the desired state of IstioInstaller.
+// The spec is a used to define a customization of the default values that are supplied with an Istio charts release.
+// It is grouped at the top level by feature, where behavior of Istio functional areas is specified.
+// Each feature contains components, where k8s resource level defaults can be overridden.
+// Because the spec is a customization API, specifying an empty InstallSpec results in a default Istio installation.
+message InstallSpec {
+    // Prefix added to all namespaces for any installed component.
+    string default_namespace_prefix = 5;
+
+    // Selection and configuration of core Istio features.
+    TrafficManagementFeatureSpec traffic_management = 25;
+    PolicyFeatureSpec policy = 26;
+    TelemetryFeatureSpec telemetry = 27;
+    SecurityFeatureSpec security = 28;
+    ConfigManagementFeatureSpec config_management = 29;
+
+    // Ingress/egress gateway installation and configuration.
+    repeated IngressGatewaySpec ingress_gateway = 30;
+    repeated EgressGatewaySpec egress_gateway = 31;
+
+    // List of external operators that will be installed (but not managed)
+    // by the IstioInstaller e.g. prometheus, grafana, kiali.
+    repeated OperatorSpec external_operators = 50;
+
+    // Global k8s defaults. Can be overridden per component.
+    KubernetesResourcesSpec k8s_defaults = 80;
+
+    // URL for the base spec. e.g.
+    //   istio-auth.yaml (looks in default dir)
+    //   file://var/istio/install/values/custom/custom-install.yaml
+    // Uses compiled in default install spec by default.
+    string base_spec_path = 100;
```

Comment:
I thought the default spec should be in install package, while the custom spec is generated by the other fields.

Reply:
I have the same q.

Reply:
ptal at the examples at the top, it may make it more clear. 
we start the install with a base profile (default profile is the default if base_spec_path is unset). 
any customizations in the InstallSpec here are ovelaid on top of the selected base InstallSpec. 
so, we could start with istio "demo" spec, but then decide we want to customize that by turning off config management:

spec:
  baseSpecPath: istio-demo.yaml
  configManagement:
    enabled: false

Reply:
ok that makes sense, essentially the base_spec_path can point to any profiles as the base.